src/FuelClient.cc

```diff
@@ -378,16 +378,31 @@ WorldIter FuelClient::Worlds(const ServerConfig &_server) const
 //////////////////////////////////////////////////
 ModelIter FuelClient::Models(const ModelIdentifier &_id)
 {
-  return const_cast<const FuelClient*>(this)->Models(_id);
+  return this->Models(_id, true);
 }
 
 //////////////////////////////////////////////////
 ModelIter FuelClient::Models(const ModelIdentifier &_id) const
 {
-  // Check local cache first
-  ModelIter localIter = this->dataPtr->cache->MatchingModels(_id);
-  if (localIter)
-    return localIter;
+  return this->Models(_id, true);
+}
+
+//////////////////////////////////////////////////
+ModelIter FuelClient::Models(const ModelIdentifier &_id, bool _checkCache)
```

Comment:
Why are both these signatures needed?  It looks like you are forwarding the non-const version to the const version.  Can the compiler not do that for us?

Reply:
The original `Models(const ModelIndentifier)` member had both a const and non-const version, so I'm doing the same for the new signature.

> Can the compiler not do that for us?

I don't think so. The `const_cast` trick is a way to avoid duplicating the code. https://stackoverflow.com/a/856839/283225